h2d/RenderContext.hx

```diff
@@ -781,7 +830,7 @@ class RenderContext extends h3d.impl.RenderContext {
 			flush();
 		var shaderChanged = needInitShaders, paramsChanged = false;
 		var objShaders = obj.shaders;
-		var curShaders = currentShaders.next;
+		var curShaders = currentShadersTail.next;

```

Comment:
think this needs a null check

```suggestion
		var curShaders = currentShadersTail?.next;
```

Reply:
No, because it's never supposed to be null. Tail points at the last global shader (`tail.next` points at object-specific shaders), and you should not be able to remove base shader from the global shader list, thus tail is never null.

Reply:
🤷 idk i had to change this to make it work locally

Reply:
Can you provide an repro sample?